### PR TITLE
Changes to allow multiple schedules of various device settings

### PR DESCRIPTION
**Pull request by lisad**
allows multiple schedules of bgTargets, insulinSensitivities and carbRatios to be uploaded for device settings, and to test that option as well as the single-schedule option.

### CODE REVIEW

**Comment by jebeck** [comment]
@lisad There are jshint issues causing the Travis build to fail here. Can you fix those? In all our repositories, you can lint with jshint using `npm run jshint` and set up a watcher to run as you work with `npm run jshint-watch`.

File: test/schema/settingsTest.js [hunk 1]
```diff
@@ -115,10 +115,36 @@ describe('schema/settings.js', function () {
...
+    helper.expectObjectField(multiCarbRatios, 'carbRatios');
+
+    it('accepts carbRatios with multiple schedules', function(done) {
+      helper.run(multiCarbRatios, done);
+    });
+
+    it('rejects carbRatios with bad internal schema', function(done) {
+      var multiCarbRatioWrong = _.cloneDeep(goodObject);
+      delete multiCarbRatioWrong.carbRatio;
+      multiCarbRatioWrong.carbRatios = { "these": "are not the droids you're looking for" };
```
Comment (jebeck) [inline test/schema/settingsTest.js]: Nice touch :)

**Comment by jebeck** [comment]
LGTM. Still want to talk about the live testing strategy and get the data model docs updated before we merge and deploy.

**Comment by jebeck** [comment]
So after our conversation yesterday, it occurs to me that if we don't have any other "live" testing of this, I'd prefer for the api ingestion tests to reflect some of the options - in particular, a nested `insulinSensitivities` with the `units.bg` of the settings object set to `mg/dL` (so the numbers for insulin sensitivity should be somewhere in the range of 60-100, for realism's sake at least). These should be converted to mmol/L for storage - basically it'd be good to see that's still happening with the nested structure... Can you add that additional test @lisad? Let me know if I'm not being clear...

**Comment by lisad** [comment]
Yep, Can do.

**Comment by jebeck** [comment]
LGTM, thanks for adding the test (and code for conversion) @lisad 

Merging.

**Pull request merged**